src/helm/proxy/clients/scale_critique_client.py

```diff
@@ -232,52 +239,77 @@ def _get_worker_responses(self, task_id: str) -> List[CritiqueResponse]:
+            #     ],
+            #     "Conciseness": [
+            #         "4"
+            #     ],
+            #     "Harmlessness": [
+            #         "5"
+            #     ],
+            #     "Factuality": [
+            #         "3"
+            #     ],
+            #     "Keyword Feedback": "Two out of five answers were not understandable"
+            #     },
+            #     {
+            #     "Helpfulness": [
+            #         "4"
+            #     ],
+            #     "Understandability": [
+            #         "4"
+            #     ],
+            #     "Completeness": [
+            #         "5"
+            #     ],
+            #     "Conciseness": [
+            #         "4"
+            #     ],
+            #     "Harmlessness": [
+            #         "5"
+            #     ],
+            #     "Factuality": [
+            #         "1"
+            #     ],
+            #     "Keyword Feedback": "typos"
             #     }
-            #   },
-            #   {
-            #     "id": "respondent_2",
-            #     "answers": {
-            #       "category_field_1": answer_1_respondent_2
-            #       "category_field_2": answer_2_respondent_2
-            #       ...
-            #     }
-            #   },
-            #   ...
             # ]
-
             # First, we get the list of respondents
-            num_respondents: int = len(annotations[list(annotations.keys())[0]])
+            num_respondents: int = len(raw_responses)
 
             # Then, we create the list of responses
             responses: List[CritiqueResponse] = []
             for respondent_index in range(num_respondents):
                 answers: Dict[str, Union[str, List[str]]] = {}
-                for field_name, field_answers in annotations.items():
-                    answers[field_name] = field_answers[respondent_index]
+                raw_response = raw_responses[respondent_index]
+                for i, (field_name, field_answers) in enumerate(raw_response.items()):
+                    if fields[i]["type"] == "category":
+                        # We need to convert the answer from an index to the actual value
+                        assert (
+                            isinstance(field_answers, list)
+                            and len(field_answers) == 1
+                            and isinstance(field_answers[0], str)
+                        )
+                        value: int = int(field_answers[0])
+                        answers[field_name] = fields[i]["choices"][value]["label"]
+                    elif fields[i]["type"] == "text":
+                        answers[field_name] = field_answers
+                    else:
+                        raise NotImplementedError(f"{fields[i]['type']} fields are not supported yet")
                 responses.append(
                     CritiqueResponse(id=str(respondent_index), respondent_id=str(respondent_index), answers=answers)
```

Comment:
Are we able to set `id` or `respondent_id` based on the actual IDs, rather than the index?

Reply:
Discussed in person: add a todo.